src/ogdc_runner/common.py

```diff
@@ -0,0 +1,322 @@
+                    value_from=models.ValueFrom(path="/tmp/published.txt"),
+                ),
+            ],
+            volume_mounts=[
+                models.VolumeMount(
+                    name=OGDC_WORKFLOW_PVC.name,
+                    mount_path="/mnt/",
+                ),
+            ],
+        )
+
+        # Set custom image if provided
+        if custom_image or custom_tag:
+            # This will override the global image setting just for this container
+            check_dir_template.image = custom_image if custom_image else None
+            check_dir_template.image_tag = custom_tag if custom_tag else None  # type: ignore[attr-defined]
+
+        with Steps(name="steps"):
+            check_dir_template()
+
+    # wait for the workflow to complete.
+    workflow_name = submit_workflow(workflow=w, wait=True)
+
+    # If overwrite is not True, we need to check the result of the
+    # `check-already-published` step to see if the data have been published or
+    # not.
+    # Check the result. Get an updated instance of the workflow, with the latest
+    # states for all notdes. Then, iterate through the nodes and find the
+    # template we define above ("check-already-published") and extract its
+    # output parameter.
+    completed_workflow = ARGO_WORKFLOW_SERVICE.get_workflow(name=workflow_name)
+    result = None
+    for node in completed_workflow.status.nodes.values():  # type: ignore[union-attr]
+        if node.template_name == "check-already-published":
+            result = node.outputs.parameters[0].value  # type: ignore[union-attr, index]
+    if not result:
+        err_msg = "Failed to check if data have been published"
+        raise OgdcWorkflowExecutionError(err_msg)
+
+    assert result in ("yes", "no")
+
+    # Cleanup this workflow, it is no longer needed
+    ARGO_WORKFLOW_SERVICE.delete_workflow(workflow_name)
+
+    return result == "yes"
+
+
+def data_already_published(
+    *,
+    recipe_config: RecipeConfig,
+    overwrite: bool,
+    custom_image: str | None = None,
+    custom_tag: str | None = None,
+) -> bool:
+    """Check for the existence of published data for the given
+    recipe and optionally remove it.
+
+    If `overwrite=True`, this function will remove any existing published data
+    for the provided recipe.
+
+    Returns `True` if data have already been published for the given recipe,
+    otherwise `False`.
+    """
+    if overwrite:
+        # If `overwrite` is True, remove the existing data and return `False`.
+        remove_existing_published_data(
+            recipe_config=recipe_config,
+            custom_image=custom_image,
+            custom_tag=custom_tag,
+        )
+        return False
+
+    return check_for_existing_published_data(
+        recipe_config=recipe_config,
+        custom_image=custom_image,
+        custom_tag=custom_tag,
+    )
+
+
+def parse_commands_from_recipe_file(recipe_dir: str, filename: str) -> list[str]:
```

Comment:
This should, I think, remain in the "recipe.simple" module. The bash style recipe file is what we are calling "simple" (and we might want to re-think the name for that too, although in a separate PR). We might have other types of recipe files that we parse for more complex use-cases in the future (e.g., exposing native argo yaml as a recipe option, which might live in `recipe.argo_yaml`).